lib/main.js

```diff
@@ -84,7 +89,8 @@ function main(options) {
     let module = JSON.parse(data);
     for (let name in module.dependencies) {
       let values = module.dependencies[name];
-      getInfo(allModules, name).want = getInstallSpec(log, name, values.version, values.from);
```

Comment:
Thank you. I'd also like to compliment you on the logical coverage in the test cases. They made it very easy to expand the functionality without having too much details on the inner workings. 

I agree. I kinda started off with wanting to keep this functionality as separate as possible, but I agree that this will keep this cleaner and more readable.